src/Config/SearchEngine.php

```diff
@@ -75,34 +85,73 @@ public function __construct(
      *
      * @return array
      */
-    public function get(): array
+    public function getConfig(): array
     {
-        $envSearchConfig = (array)$this->stageConfig->get(DeployInterface::VAR_SEARCH_CONFIGURATION);
-
-        if ($this->isSearchConfigValid($envSearchConfig)
-            && !$this->configMerger->isMergeRequired($envSearchConfig)
-        ) {
-            return $this->configMerger->clear($envSearchConfig);
+        if ($this->config === null) {
+            $resolveMerge = function () {
+                $searchConfig = (array)$this->stageConfig->get(DeployInterface::VAR_SEARCH_CONFIGURATION);
+
+                if (isset($searchConfig['engine']) && !$this->configMerger->isMergeRequired($searchConfig)) {
+                    return $this->configMerger->clear($searchConfig);
+                }
+
+                return $this->configMerger->mergeConfigs(
+                    $this->getSearchConfig(),
+                    $searchConfig
+                );
+            };
```

Comment:
What configurations will be set in .env.yaml file has 

> SEARCH_CONFIGURATION:            
>                 elasticsearch_server_hostname: hostname    
>                 elasticsearch_server_port: '123456' 
>                 elasticsearch_index_prefix: magento 
>                 elasticsearch_server_timeout: '15'  

so, no **engine** and no **_merge**
Seems like these configs will be ignored

Reply:
It is current behavior, you think it should be changed?

Reply:
It should be changed, and we may validate this config before instead of changing it here (validation is failed when configs exist, but engine was not provided) . 
You may move it to separate task - I'm OK with it.

Reply:
Created https://magento2.atlassian.net/browse/MAGECLOUD-3470